crates/turbopack-core/src/chunk/optimize.rs

```diff
@@ -3,310 +3,30 @@
 //! Usually chunks are optimized by limiting their total count, restricting
 //! their size and eliminating duplicates between them.
 
-use std::{cell::RefCell, mem::take, rc::Rc};
-
 use anyhow::Result;
-use indexmap::{IndexMap, IndexSet};
-use turbo_tasks::TryJoinIterExt;
-use turbo_tasks_fs::{FileSystemPathOptionVc, FileSystemPathVc};
-
-use super::{ChunkVc, ChunksVc};
-use crate::{
-    asset::{Asset, AssetVc},
-    chunk::Chunk,
-};
-
-/// A functor to optimize a set of chunks.
-#[turbo_tasks::value_trait]
-pub trait ChunkOptimizer {
-    fn optimize(&self, chunks: ChunksVc) -> ChunksVc;
-}
-
-/// Trait to mark a chunk as optimizable.
-#[turbo_tasks::value_trait]
-pub trait OptimizableChunk: Chunk + Asset {
-    fn get_optimizer(&self) -> ChunkOptimizerVc;
-}
```

Comment:
This is no longer relevant, as the chunking context has to know about all the different chunk types it can handle anyway. So instead of the chunks themselves implementing this trait, it's the chunking context which resolves chunks to particular types, and optimizes them accordingly.